main.py

```diff
@@ -50,44 +50,39 @@ def _run(self):
                 continue
 
             try:
-                artist_name = TitleParser.get_artist_name_from_submission_title(submission.title)
+                title = TitleParser.create_parsed_submission_from_post_title(submission.title)
             except LookupError:
                 # Can't find artist from submission name, skipping
                 self.database.save_submission(submission)
                 continue
-            
-            # Get artist for most future use
-            prio_artist = TitleParser.get_prioritized_artist(artist_name)
-
-            # Check Youtube.
-            youtube_service = self.youtube.get_service_result(submission.url)
-            if youtube_service.exceeds_threshold is True:
-                self._perform_exceeds_threshold_mod_action(submission, youtube_service)
-                self.database.save_submission(submission)
-                continue
-            
-            # Check Soundcloud.
-            soundcloud_service = self.soundcloud.get_service_result(submission.url)
-            if soundcloud_service.exceeds_threshold is True:
-                self._perform_exceeds_threshold_mod_action(submission, soundcloud_service)
-                self.database.save_submission(submission)
-                continue
 
-            # Check Last.FM
-            try:
-                last_fm_service = self.last_fm.get_service_result(prio_artist)
-                if last_fm_service.exceeds_threshold is True:
-                    self._perform_exceeds_threshold_mod_action(submission, last_fm_service)
-                    self.database.save_submission(submission)
-                    continue
-            except pylast.WSError:
-                # Go ahead and continue execution.  Don't want to fail completely just because one service failed.
-                pass
+            for service in self.services:
+                try:
```

Comment:
Once we get to this point we are always going to add the submission to the DB.  Perhaps using a finally clause here and dropping the submit functionality from  if result.exceeds_threshold is True (https://github.com/martijnboers/BlotterTrax/pull/38/files#diff-5bc02cefb3ea9e27f1a6776eabd1935dR66) and the successful block (https://github.com/martijnboers/BlotterTrax/pull/38/files#diff-5bc02cefb3ea9e27f1a6776eabd1935dR87)

Reply:
I think I got what you mean, could you verify that my solution is what you intended with this comment?

Reply:
Yep!  LGTM.